syncd/syncd_reinit.cpp

```diff
@@ -748,6 +748,182 @@ void helperCheckPriorityGroupsIds()
+
+    sai_status_t status = sai_scheduler_group_api->get_scheduler_group_attribute(schedulerGroupId, 1, &attr);
+
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to get child count for scheduler group: 0x%llx", schedulerGroupId);
+
+        exit_and_notify(EXIT_FAILURE);
+    }
+
+    return attr.value.u32;
+}
+
+std::vector<sai_object_id_t> saiGetSchedulerGroupChildList(sai_object_id_t schedulerGroupId)
+{
+    SWSS_LOG_ENTER();
+
+    uint32_t childCount = saiGetSchedulerGroupChildCount(schedulerGroupId);
+
+    std::vector<sai_object_id_t> childs;
+
+    childs.resize(childCount);
+
+    sai_attribute_t attr;
+
+    attr.id = SAI_SCHEDULER_GROUP_ATTR_CHILD_LIST;
+    attr.value.objlist.list = childs.data();
+    attr.value.objlist.count = (uint32_t)childs.size();
+
+    sai_status_t status = sai_scheduler_group_api->get_scheduler_group_attribute(schedulerGroupId, 1, &attr);
+
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to get child list for scheduler group: 0x%llx", schedulerGroupId);
+
+        exit_and_notify(EXIT_FAILURE);
+    }
+
+    childs.resize(attr.value.objlist.count);
+
+    SWSS_LOG_DEBUG("got %zu childs on scheduler group 0x%llx", childs.size(), schedulerGroupId);
+
+    return childs; // scheduler groups + queues
+}
+
+std::set<sai_object_id_t> g_defaultSchedulerGroupsRids;
+
+void helperCheckSchedulerGroupsIds()
+{
+    SWSS_LOG_ENTER();
+
+    for (const auto& portId: saiGetPortList())
+    {
+        // each group can contain next scheduler group or queue
+
+        for (const auto& schedGroupId: saiGetSchedulerGroupList(portId))
+        {
+            // create entry in asic view if missing
+            // we assume here that SchedulerGroups numbers will
+            // not be changed during restarts
+
+            redisCreateDummyEntryInAsicView(schedGroupId);
+
+            g_defaultSchedulerGroupsRids.insert(schedGroupId);
+
+            // childs may be scheduler group or queue
+
+            for (const auto& childId: saiGetSchedulerGroupChildList(schedGroupId))
+            {
+                sai_object_type_t objectType = sai_object_type_query(childId);
+
+                if (objectType == SAI_OBJECT_TYPE_QUEUE)
+                {
+                    // skip queue since we are not interested here
+                }
+                else if (objectType == SAI_OBJECT_TYPE_SCHEDULER_GROUP)
+                {
+                    redisCreateDummyEntryInAsicView(childId);
+
+                    g_defaultSchedulerGroupsRids.insert(childId);
+                }
+                else
+                {
+                    SWSS_LOG_ERROR(
+                            "unexpected object type %s on scheduler group child list 0x%llx",
+                            sai_serialize_object_type(objectType).c_str(),
+                            schedGroupId);
+
+                    exit_and_notify(EXIT_FAILURE);
+                }
+            }
```

Comment:
this seems unnecessary, the childlist scheduler groups should be include in the get schedulergroup list